ipn/ipnlocal/local.go

```diff
@@ -2674,7 +2680,9 @@ func (b *LocalBackend) WatchNotifications(ctx context.Context, mask ipn.NotifyWa
 		select {
 		case <-ctx.Done():
 			return
-		case n, ok := <-ch:
+		case <-session.closeCh:
+			return
```

Comment:
Nit: we can avoid adding a new channel just for the purpose of closing by re-using the `ctx.Done()` which serves the same purpose. 

Instead of having a `closeCh` on line 164 above, it can instead be a `cancel context.CancelFunc` that `serve.Go` calls.

Reply:
SGTM. Creating a cancellable context still means adding a new channel under the hoods, but we already do that in certain cases:
https://github.com/tailscale/tailscale/blob/efbb36f1d0d37b1d8bb2a674cc8b6d9a583379eb/ipn/ipnlocal/local.go#L2668-L2672
So we surely could create it in an outer scope and call its `context.CancelFunc` from `serve.go`.